app/export.go

```diff
@@ -19,7 +19,13 @@ func (app *App) ExportAppStateAndValidators(forZeroHeight bool, jailAllowedAddrs
 		return servertypes.ExportedApp{}, err
 	}
 
-	app.InitializeAppVersion(ctx)
+	versionToInit := v1
+	if app.LastBlockHeight() >= app.upgradeHeightV2 {
+		versionToInit = v2
+	}
```

Comment:
This won't work if the user hasn't set the appropriate flag. I would prefer trying to read the consensus params to know what the version is